#167918277 Implement Create Category Functionality

henryade opened the pull request:
#### What does this PR do?
Enables Admin and Superadmin user create categories

#### Description of Task proposed in this pull request?
- Update existing model with 'description' attribute
- Build the create category endpoint
- Write end to end test

#### How should this be manually tested (Quality Assurance)?
- Make a `POST` request to `/api/v1/categories/create`
- A `name` and `description` field is to be present in the request body.

#### What are the relevant pivotal tracker stories?
- Story ID:  [#167918277](https://www.pivotaltracker.com/story/show/167918277)

#### Any background context you want to add (Operations Impact)?
- Run `npm run db:rollback` to migrate all tables

#### What I have learned working on this feature: 
- I learnt how to validate fields

#### Screenshots: 
Create Category:
<img width="1440" alt="Screenshot 2019-08-15 at 12 25 46 PM" src="https://user-images.githubusercontent.com/47576037/63092037-45c3d500-bf58-11e9-9d9f-aaf95a90add4.png">

Resending the Same Request:
<img width="1440" alt="Screenshot 2019-08-15 at 12 25 53 PM" src="https://user-images.githubusercontent.com/47576037/63092054-4f4d3d00-bf58-11e9-9cf5-37e2c781e48d.png">

chialuka commented on server/routes/category.js:
```diff
@@ -0,0 +1,16 @@
+route.post(
+  '/create',
+  verifyToken,
+  authorizeUser(4),
```
Don't admins and super-admins have sessions that should be checked?

henryade replied:
they do. Thanks for pointing this out. I have added this.

dinobi reviewed (commented):
Can you include a model test

henryade committed:
feature-[167918277]: implement create category functionality
- update existing model with 'description' attribute
- build the create category endpoint
- write end to end test
[Delivers #167918277]
